src/core/components/FileUpload.jsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import withStyles from '@material-ui/core/styles/withStyles';
+import Button from './CustomButton';
+import { validateFile ,validateFileSize } from '../utils/utils';
+
+const styles = ({
+  paragrafo: {
+    margin: 0
+  }
+})
+
+class FileUpload extends React.Component {
```

Comment:
Aqui também poderia ser um functional component. https://reactjs.org/docs/components-and-props.html#function-and-class-components

Reply:
esse aqui mantem o estado do input

Reply:
Certeza? Não vejo this.state sendo usado, mas pode ser assim, se é necessário.